[export] Add TracingContext

angelayi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #149295
* __->__ #149294

TracingContext is added to all tracing locations -- in torch.export this is where we call make_fx (for training IR) and aot_export_module (for inference IR), and in run_decompositions where we call aot_export_module

Differential Revision: [D71298927](https://our.internmc.facebook.com/intern/diff/D71298927)

angelayi committed:
[export] Add TracingContext

[ghstack-poisoned]

angelayi committed:
Update on "[export] Add TracingContext"




[ghstack-poisoned]

angelayi commented:
@angelayi has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D71298927).

ydwu4 reviewed (approved):
The places that we add tracing context seem random to me. Any principle behind? Can wrote a note somewhere either in the code or in the comment section.

angelayi commented:
@angelayi has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D71298927).

angelayi committed:
Update on "[export] Add TracingContext"


TracingContext is added to all tracing locations -- in torch.export this is where we call make_fx (for training IR) and aot_export_module (for inference IR), and in run_decompositions where we call aot_export_module

Differential Revision: [D71298927](https://our.internmc.facebook.com/intern/diff/D71298927)

[ghstack-poisoned]

angelayi commented:
@angelayi has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D71298927).
